garden-backend-service/src/models/base.py

```diff
@@ -29,16 +29,27 @@ async def get(
     async def get_or_create(
```

Comment:
Hot take: since we only actually use the `get_or_create` helper in the authed_user dependency, do we really need it our base object? This method is starting to feel specific to the authed_user usage of it (for example, is it always unlikely that the object might have disappeared? or is that only a reasonable assumption for now because we never delete from the users table?), and I wonder if it might be easier to reason about if we just pulled the integrity handling logic out to the dependency itself and cut our losses on the helper method.

I also am not totally seeing how this manages to avoid the race condition, though I'm not 100% sure I see where the race condition came from in the first place. I think I'm missing some ORM subtlety most likely